tool/tsh/tsh.go

```diff
@@ -1388,11 +1388,12 @@ func onLogin(cf *CLIConf) error {
 			if err := tc.SaveProfile(cf.HomePath, true); err != nil {
 				return trace.Wrap(err)
 			}
-			if err := updateKubeConfig(cf, tc, ""); err != nil {
-				return trace.Wrap(err)
+			if err := updateKubeConfig(cf, tc, ""); err == nil {
+				return trace.Wrap(onStatus(cf))
 			}
 
-			return trace.Wrap(onStatus(cf))
+			// If we fail to update the kube config, we may be switching to
```

Comment:
This comment looks a little weird here with the indentation, but LGTM otherwise.